lisp/pdf-view.el

```diff
@@ -922,8 +922,9 @@ See also `pdf-view-use-imagemagick'."
 
 (defun pdf-view-use-scaling-p ()
   "Return t if scaling should be used."
-  (and (memq (pdf-view-image-type)
-             '(imagemagick image-io))
+  (and (or (and (eq (framep-on-display) 'ns) (string-equal emacs-version "27.0.50"))
```

Comment:
The version checking is not perfect here.

Reply:
It might be worth looking into image-transforms-p rather than relying on the version number? Something like this:

(and (eq (framep-on-display) 'ns) (fboundp image-transforms-p))